### PR TITLE
Include 'highlight.js' for Syntax Highlighting

### PR DESCRIPTION
**Pull request by xsc**
This includes `highlight.js` with the `clojure` and `clojure-repl` language packs, as well as `solarized-light` as its syntax highlighting theme.

See #135.

### CODE REVIEW

**Comment by xsc** [comment]
I noticed that Pegdown already generates the code blocks compatible to `highlight.js` – no idea how I thought otherwise in the first place.

We could also include the highlight.js assets directly, if desired, although this would increase the size of the codox JAR, plus we'd need to settle on a list of supported languages to bundle.

**Comment by weavejester** [comment]
We need to include the highlight.js assets directly, because people could be reading API documentation offline. That's why I've included jQuery rather than use a CDN.

Rather than make highlight.js directly configurable, I think it makes more sense to provide a set of defaults that will work for 90% of the projects out there, and allow HTML transformations to cover any edge cases. So `clojure` and `clojure-repl` as the syntax highlighting libraries, and anyone who wants more can always add the additional CSS assets directly.

**Comment by weavejester** [comment]
Does Pegdown work out-of-the-box with highlight.js, then?

**Comment by xsc** [comment]
> Does Pegdown work out-of-the-box with highlight.js, then?

It does.

> We need to include the highlight.js assets directly, because people could be reading API documentation offline. That's why I've included jQuery rather than use a CDN.

I'm wondering whether this is necessary for syntax highlighting – which is nice to have but it's not the end of the world if missing.

> So `clojure` and `clojure-repl` as the syntax highlighting libraries, and anyone who wants more can always add the additional CSS assets directly.

Do you also want this to apply to the syntax highlighting themes (solarized dark, solarized light, etc...)? I.e. do we just bundle the JS language packs and call the highlight.js hook – but adding the desired syntax highlighting CSS is left to the user? Or do we include all currently available themes with codox?

I'm happy to implement both and I think it even makes sense to let themes choose the syntax highlighting CSS.

**Comment by weavejester** [comment]
> I'm wondering whether this is necessary for syntax highlighting – which is nice to have but it's not the end of the world if missing.

I think docs should ideally have the same formatting regardless of whether they're read online or offline, even if that formatting is superficial. It's something that I think would really bug me if I was reading Codox docs on a plane :)

> Do you also want this to apply to the syntax highlighting themes (solarized dark, solarized light, etc...)? I.e. do we just bundle the JS language packs and call the highlight.js hook – but adding the desired syntax highlighting CSS is left to the user? Or do we include all currently available themes with codox?

We pick one theme that works well with the current CSS. Let's go with solarized light for now. The syntax highlighting CSS should be considered to be part of the overall theme, IMO.

Regarding the JS language packs, let's just include the Clojure ones for now.

**Comment by xsc** [comment]
I updated the PR description to match the latest commits. Once this is through, I'll tackle the actual theme support.

**Comment by weavejester** [comment]
I think you can remove the highlight.js README.md file.

File: codox/src/codox/writer/html.clj [hunk 1]
```diff
@@ -433,6 +436,8 @@
     (copy-resource "codox/css/default.css" "css/default.css")
     (copy-resource "codox/js/jquery.min.js" "js/jquery.min.js")
     (copy-resource "codox/js/page_effects.js" "js/page_effects.js")
+    (copy-resource "codox/highlight/highlight.pack.js" "js/highlight.js")
```
Comment (weavejester) [inline codox/src/codox/writer/html.clj]: Maybe this should be `hightlight.min.js` to match the jQuery form.

**Comment by xsc** [comment]
Removed README, addressed the comment.

File: codox/src/codox/writer/html.clj [hunk 2]
```diff
@@ -433,6 +436,8 @@
     (copy-resource "codox/css/default.css" "css/default.css")
     (copy-resource "codox/js/jquery.min.js" "js/jquery.min.js")
     (copy-resource "codox/js/page_effects.js" "js/page_effects.js")
+    (copy-resource "codox/highlight/highlight.min.js" "js/highlight.js")
```
Comment (weavejester) [inline codox/src/codox/writer/html.clj]: Should probably be copied to `js/highlight.min.js` to match `js/jquery.min.js`.

Comment (xsc) [inline codox/src/codox/writer/html.clj]: Adjusted in 151b9d1.

**Comment by weavejester** [comment]
Could you also add in some syntax-highlighted code to the example project?

Perhaps in `example/src/clojure/codox/markdown.clj` in order to test syntax highlighting within a docstring, and also in `example/doc/02-formatting.md` to test syntax highlighting in a doc file.

**Comment by weavejester** [comment]
I think once the example project is updated we should be good to merge.

**Comment by xsc** [comment]
The existing code blocks already had syntax highlighting (due to `highlight.js`'s language detection, I presume) – I added fenced blocks for good measure.

**Comment by weavejester** [comment]
Perfect, thanks!

**Pull request merged**